### PR TITLE
[integer] Add type declaration file

### PR DESCRIPTION
**Pull request by Morfent**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not provide its own types, and you can not add them.
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, and `strictNullChecks` set to `true`.

### CODE REVIEW

**Comment by Morfent** [comment]
@JoshuaWise, could you take a quick look at this in case I missed anything?

File: types/integer/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+// This can't just be written as a simple class since Integer instances can't
+// be constructed by its actual constructor without throwing, but calling it as
+// a regular function constructs new instances just fine...
+declare function Integer(val: Integer.IntLike): Integer.IntClass;
+
+declare namespace Integer {
+	type IntLike = number | string | IntClass;
+
+	function fromBits(low: number, high?: number): IntClass;
+	function fromNumber(val: number, defaultVal?: IntClass): IntClass;
```
Comment (JoshuaWise) [inline types/integer/index.d.ts]: `defaultVal?` here can be a `number` or `IntClass`.

File: types/integer/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+	function fromString(val: string, radix?: number): IntClass;
```
Comment (JoshuaWise) [inline types/integer/index.d.ts]: There's an optional third argument (to specify a default value) which can either be a `string` or `IntClass`

File: types/integer/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+		neq(val: IntLike): boolean;
+		notEquals(val: IntLike): boolean;
+		isNotEqualTo(val: IntLike): boolean;
+		doesNotEqual(val: IntLike): boolean;
+
+		gt(val: IntLike): boolean;
+		greaterThan(val: IntLike): boolean;
+		gte(val: IntLike): boolean;
+		greaterThanOrEquals(val: IntLike): boolean;
```
Comment (JoshuaWise) [inline types/integer/index.d.ts]: This is missing `isGreaterThan()` and `isGreaterThanOrEqualTo()`

File: types/integer/index.d.ts [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+
+		lt(val: IntLike): boolean;
+		lessThan(val: IntLike): boolean;
+		lte(val: IntLike): boolean;
+		lessThanOrEquals(val: IntLike): boolean;
```
Comment (JoshuaWise) [inline types/integer/index.d.ts]: This is missing `isLessThan()` and `isLessThanOrEqualTo()`

**Review by JoshuaWise — CHANGES_REQUESTED** [review]

**Comment by Morfent** [comment]
Made the changes you brought up. Thanks for catching them before they caused any problems

File: types/integer/index.d.ts [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+
+		toNumber(): number;
+		toNumberUnsafe(): number;
+		toString(radix?: number): string;
+		valueOf(): number;
+	}
+}
+
+export = Integer;
+export as namespace Integer;
```
Comment (RyanCavanaugh) [inline types/integer/index.d.ts]: This is not a UMD module, so there shouldn't be a UMD module declaration.

**Review by RyanCavanaugh — CHANGES_REQUESTED** [review]

File: types/integer/index.d.ts (hunk 5 above)
Comment (Morfent) [inline types/integer/index.d.ts]: Would `export default` be more appropriate?

Comment (RyanCavanaugh) [inline types/integer/index.d.ts]: No.
```
C:\Throwaway>node
> require('integer').default
undefined
>
```
You can just remove line 111.

Comment (Morfent) [inline types/integer/index.d.ts]: ...checking the documentation again, you're right

**Comment by Morfent** [comment]
Removed line 11

**Review by JoshuaWise — APPROVED** [review]

**Comment by RyanCavanaugh** [comment]
@RyanCavanaugh - Thanks for your review of this PR! Can you please look at the new code and update your review status if appropriate?

**Pull request merged**